hera_cal/utils.py

```diff
@@ -1244,14 +1248,18 @@ def red_average(data, reds=None, bl_tol=1.0, inplace=False,
                 n = np.asarray([data.get_nsamples(bl + (pol,)) for bl in blg])
                 tint = np.asarray([data.integration_time[data.antpair2ind(bl + (pol,))] for bl in blg])[:, :, None]
                 w = np.asarray([wgts[bl + (pol,)] for bl in blg]) * tint
-
             # take the weighted average
             wsum = np.sum(w, axis=0).clip(1e-10, np.inf)  # this is the normalization
             davg = np.sum(d * w, axis=0) / wsum  # weighted average
-            navg = np.sum(n * f, axis=0)         # this is the new total nsample (without flagged elements)
             fmax = np.max(f, axis=2)             # collapse along freq: marks any fully flagged integrations
             iavg = np.sum(tint.squeeze() * fmax, axis=0) / np.sum(fmax, axis=0).clip(1e-10, np.inf)
-            favg = np.isclose(wsum, 0.0)         # this is getting any fully flagged pixels
+            binary_wgts = (~np.isclose(w, 0)).astype(np.float)  # binary weights.
+            if propagate_flags:
+                fsum = np.sum(n * binary_wgts * f, axis=0)
```

Comment:
I don't think `fsum` here should have n or binary weights, right? Shouldn't this whole block just be
```
binary_wgts = (~np.isclose(w, 0)).astype(np.float)  # binary weights.
navg = np.sum(n * binary_wgts, axis=0)
if propagate_flags:
    favg = np.all(np.isclose(w * f, 0), axis=0)
else:
    favg = np.all(np.isclose(w, 0), axis=0)
```

I could be wrong though, lmk

Reply:
I agree that this makes sense and is more consistent with the operating philosophy of `red_average`. If people want `nsamples=0` to be flagged ,they can add those flags before calling the function.